server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/newcode/NewCodeApi.java

```diff
@@ -66,6 +66,7 @@ public CompletableFuture<NewCodeDefinition> getNewCodeDefinition(String projectK
       return CompletableFuture.completedFuture(NewCodeDefinition.withNumberOfDays(days, date));
     }
     if (modeString.equalsIgnoreCase("PREVIOUS_VERSION")) {
+      parameter = periodFromWs.hasParameter() ? parameter : null;
```

Comment:
For now, I only encountered an issue with the `PREVIOUS_VERSION` definition, ideally, we could handle null for each case. Shall we just fall back to `Since DATE` for each scenario?